includes/buddypress-group-email-subscription.php

```diff
@@ -12,15 +12,30 @@ function hcommons_remove_bpges_actions() {
 	remove_action( 'ass_digest_event', 'ass_daily_digest_fire' );
 	remove_action( 'ass_digest_event_weekly', 'ass_weekly_digest_fire' );
 }
+
 add_action( 'bp_init', 'hcommons_remove_bpges_actions' );
 
+// Spark doesn't like random emails that are not on it's white list.
+// Welcome emails (on join group uses the first admin as the from email which errors out Spark.
+// We replace this from email with a noreply email.
+add_action( 'bp_email_set_from', function( $from, $email_address, $name, $email_type) {
+	if ( $email_type->get('type') === "bp-ges-welcome" ) {
+		$array = explode(".", network_site_url());
```

Comment:
is there a better function the network_site_url? If we can get that parser out the better for us. maybe wp_site_url (wont pick up alias sites though)?

Reply:
I looked in detail at emails from society groups and the from domain is the same for all societies, we do not want society specific group welcome from domains.
This appears to be the most common approach to getting the from email domain.
`                $sitename = strtolower( $_SERVER['SERVER_NAME'] );
                if ( 'www.' == substr( $sitename, 0, 4 ) ) {
                        $sitename = substr( $sitename, 4 );
                }
`
I'm closing this pull request. There are unrelated format changes in the pull request.